src/databricks/labs/lakebridge/resources/assessments/synapse/monitoring_metrics_extract.py

```diff
@@ -0,0 +1,160 @@
+
+        artifacts_client = get_synapse_artifacts_client(synapse_workspace_settings)
+        workspace = SynapseWorkspace(workspace_tz, artifacts_client)
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+        metrics_client = get_azure_metrics_query_client()
+        synapse_metrics = SynapseMetrics(metrics_client)
+
+        workspace_info = workspace.get_workspace_info()
+        print(workspace_info)
+
+        if "id" not in workspace_info:
+            raise ValueError("ERROR: Missing Workspace ID for extracting Workspace Level Metrics")
+        workspace_resource_id = workspace_info["id"]
+        logger.info(f"workspace_resource_id  →  {workspace_resource_id}")
+        metrics_df = synapse_metrics.get_workspace_level_metrics(workspace_resource_id)
+        insert_df_to_duckdb(metrics_df, db_path, "workspace_level_metrics")
+
+        # SQL Pool Metrics
+
+        exclude_dedicated_sql_pools = synapse_profiler_settings.get("exclude_dedicated_sql_pools", None)
+        dedicated_sql_pools_profiling_list = synapse_profiler_settings.get("dedicated_sql_pools_profiling_list", None)
+
+        logger.info(f" exclude_dedicated_sql_pools        →  {exclude_dedicated_sql_pools}")
+        logger.info(f" dedicated_sql_pools_profiling_list →  {dedicated_sql_pools_profiling_list}")
+
+        if exclude_dedicated_sql_pools:
+            logger.info(
+                f" exclude_dedicated_sql_pools is set to {exclude_dedicated_sql_pools}, Skipping metrics extract for Dedicated SQL pools"
+            )
+        else:
+            dedicated_sqlpools = workspace.list_sql_pools()
+            all_dedicated_pools_list = [pool for poolPages in dedicated_sqlpools for pool in poolPages]
+            dedicated_pools_to_profile = (
+                all_dedicated_pools_list
+                if not dedicated_sql_pools_profiling_list
+                else [pool for pool in all_dedicated_pools_list if pool['name'] in dedicated_sql_pools_profiling_list]
+            )
+            msg = f"Pool names to extract metrics: {[entry['name'] for entry in dedicated_pools_to_profile]}"
+            logger.info(msg)
+
+        if exclude_dedicated_sql_pools:
+            logger.info(
+                f" exclude_dedicated_sql_pools is set to {exclude_dedicated_sql_pools}, Skipping metrics extract for Dedicated SQL pools"
+            )
+        else:
+            pools_df = pd.DataFrame()
+            for idx, pool in enumerate(dedicated_pools_to_profile):
+                pool_name = pool['name']
+                pool_resoure_id = pool['id']
+
+                logger.info(f"{'*'*70}")
+                logger.info(f"{idx+1}) Pool Name: {pool_name}")
+                logger.info(f"   Resource id: {pool_resoure_id}")
+                print(f"{idx+1}) Pool Name: {pool_name}")
+                print(f"   Resource id: {pool_resoure_id}")
+
+                pool_metrics_df = synapse_metrics.get_dedicated_sql_pool_metrics(pool_resoure_id)
```

Comment:
This throws a `BadRequest` exception while fetching `DWULimit` metric which DNE. Only `BigDataPoolApplicationsEnded`, `BigDataPoolAllocatedCores`, `BigDataPoolAllocatedMemory`, `BigDataPoolApplicationsActive` are available.